### DISCUSSION TITLE
Does pyinstaller can't build bin with python3 and then run in python2 server?

### DISCUSSION THREAD

**Discussion by JulianZhang** [pyinstaller/pyinstaller#7681] [Help] [answered]
I had build our bin with pyinstaller with python3， the binary working fine in any server who had python3.
But when I copy it to a server only have python2，I will get error as blow.

`/usr/local/python3/bin/python3: bad interpreter: No such file or directory
`
I want to know  does the pyinstaller support run binary with out the python3 interpreter?

**Comment by rokm** [comment, marked as answer]
PyInstaller does not use the system-installed interpreter - it sets up isolated embedded interpreter and runs your code in it.

So if you are seeing that message, most likely your code (or 3rd party code that you are using) is trying to run a subprocess with system-installed python interpreter.

**Reply by JulianZhang** [reply]
Thanks
